### PR TITLE
move counsel search commands list to variable. Help with extend other…

### PR DESCRIPTION
**Pull request by ivasonn**
Move counsel search commands list to variable. Help with extend other commands (for example spacemacs/counsel-search)

### CODE REVIEW

File: ivy.el [hunk 1]
```diff
@@ -253,6 +253,12 @@ It is a list of (CALLER . HEIGHT).  CALLER is a caller of
 If `(minibuffer-depth)' equals this, `ivy-completing-read' will
 act as if `ivy-completing-read-handlers-alist' is empty.")
 
+(defvar counsel--search-commands '(counsel-git-grep
+                                   counsel-ag
+                                   counsel-rg
+                                   counsel-pt)
+  "counsel search commands")
```
Comment (basil-conto) [inline ivy.el]: Please write full sentences, where the first non-keyword word is capitalised and the sentence ends in a full stop.  How about the following:
```el
"List of Counsel search commands."
```

File: ivy.el [hunk 2]
```diff
@@ -253,6 +253,12 @@ It is a list of (CALLER . HEIGHT).  CALLER is a caller of
 If `(minibuffer-depth)' equals this, `ivy-completing-read' will
 act as if `ivy-completing-read-handlers-alist' is empty.")
 
+(defvar counsel--search-commands '(counsel-git-grep
```
Comment (basil-conto) [inline ivy.el]: Is there any reason why this variable is defined so far away from its call site? What is this variable's purpose? What other use-cases for this variable are there? Wouldn't it be safer to initialise this variable to `nil` and have something like
```el
(cl-pushnew #'counsel-git-grep counsel--search-commands)
```
after every search command is defined?

**Comment by ivasonn** [comment]
1) That's my fault.
2) make more customizable, without overriding whole function such as ivy--highlight-default
3) i am not looking so far, but store commands in variable more correctly. For example after appending new command (like spacemacs did), we don't need edit source code or override func (how i said in previous comment)
4) i'm agree. it's better.

File: ivy.el (hunk 2 above)
Comment (ivasonn) [reply ivy.el]: 1) That's my fault. 
2) make more customizable, without overriding whole function such as ivy--highlight-default
3) i am not looking so far, but store commands in variable more correctly. For example after appending new command (like spacemacs did), we don't need edit source code or override func (how i said in previous comment)
4) i'm agree. it's better.

File: ivy.el [hunk 3]
```diff
@@ -253,6 +253,9 @@ It is a list of (CALLER . HEIGHT).  CALLER is a caller of
 If `(minibuffer-depth)' equals this, `ivy-completing-read' will
 act as if `ivy-completing-read-handlers-alist' is empty.")
 
+(defvar counsel--search-commands nil
+  "List of Counsel search commands.")
```
Comment (basil-conto) [inline ivy.el]: Sorry, I didn't realise I was looking at `ivy.el`; I thought this was in `counsel.el`.

Given that this is being used in `ivy.el`, and given that its intented purpose is as an external API, it has to be given the `ivy-` (not `ivy--`) prefix and a name more related to its purpose, e.g. `ivy-highlight-search-commands` (I'm not very good at naming things, perhaps @abo-abo has a better suggestion).

**Comment by abo-abo** [comment]
Thanks, I've made some stylistic changes. Please review.

**Comment by ivasonn** [comment]
thanks.

**Pull request closed without merging**